analyze_test.go

```diff
@@ -2427,3 +2427,88 @@ func TestAnalyze_headers_more(t *testing.T) {
 		})
 	}
 }
+
+//nolint:funlen
+func TestAnalyze_matchFn(t *testing.T) {
+	t.Parallel()
+	// define a test matchFn so that "state" directive has different definition in OSS and matchFn
+	testDirectiveMap := map[string][]uint{
+		"state":    {ngxAnyConf | ngxConfTake2},
+		"test_dir": {ngxAnyConf | ngxConfTake1},
+	}
+	testMatchFn := func(directive string) ([]uint, bool) {
+		masks, matched := testDirectiveMap[directive]
+		return masks, matched
+	}
+
+	testcases := map[string]struct {
+		stmt    *Directive
+		ctx     blockCtx
+		wantErr bool
+	}{
+		"only in matchFn ok": {
+			&Directive{
+				Directive: "test_dir",
+				Args:      []string{"args1"},
+				Line:      5,
+			},
+			blockCtx{"http", "upstream"},
+			false,
+		},
+		"in matchFn and OSS but only satisfy matchFn ok": {
```

Comment:
It's taking me a bit to understand these subtest names. I think "in matchFn and Oss..." means that the directive is matched by both `testMatchFn` and the existing built-in map? I think over time this name might lose meaning as we continue refactoring the way the maps are arranged and directives are "enabled".

When these tests fail for some reason, Go is going to print the test name as something like: `TestAnalyze_matchFn/in_matchFn_and_OSS_but_only_satisfy matchFn_ok` making it a bit harder for people to find the offending subtest as `cmd+F` is not going to work. Consider naming these tests to match the output to make things a little easier.

```suggestion
		"MatchedByFuncAndBuiltinMap": {
```

Reply:
Sure! Thanks for your suggestion!

Reply:
I modify "OSS" to "default map" since in the future this will change to a default behavior when we don't pass matchFn for OSS and N+, feel free to provide any further suggestions

Reply:
Unresolving because I don't think it's been improved substantially and subtest names still include spaces.

Reply:
I've taken a look at [subtest document](https://go.dev/blog/subtests), add some annotations and change the names, thanks for suggestions.